src/Types/ContractAddress.cs

```diff
@@ -20,10 +23,47 @@ public sealed record ContractAddress(ulong Index, ulong SubIndex) : IAddress
 
     internal static ContractAddress From(Grpc.V2.ContractAddress contractAddress) => new(contractAddress.Index, contractAddress.Subindex);
 
+    /// <summary>
+    /// Byte size of <see cref="ContractAddress"/>.
+    /// </summary>
+    public const uint BytesLength = 16;
+
     /// <summary>
     /// Converts the contract address to its corresponding protocol buffer message instance.
     ///
     /// This can be used as the input for class methods of <see cref="Client.RawClient"/>.
     /// </summary>
     public Grpc.V2.ContractAddress ToProto() => new() { Index = this.Index, Subindex = this.SubIndex };
+
+    /// <summary>
+    /// Attempt to deserialize a contract address from a span of bytes.
+    /// </summary>
+    /// <param name="bytes">The span of bytes.</param>
+    /// <param name="output">Where to write the result of the operation.</param>
+    public static bool TryDeserial(ReadOnlySpan<byte> bytes, out (ContractAddress? Address, string? Error) output)
+    {
+        if (bytes.Length < BytesLength)
+        {
+            var msg = $"Invalid length of input in `ContractAddress.TryDeserial`. Expected at least {BytesLength} bytes, found {bytes.Length}";
+            output = (null, msg);
+            return false;
+        };
+
+        var index = BinaryPrimitives.ReadUInt64BigEndian(bytes);
+        var subindex = BinaryPrimitives.ReadUInt64BigEndian(bytes);
+
+        output = (new ContractAddress(index, subindex), null);
+        return true;
+    }
+
+    /// <summary>
+    /// Serialize the ContractAddress in big-endian format.
+    /// </summary>
+    public byte[] ToBytes()
+    {
+        using var memoryStream = new MemoryStream((int)BytesLength); // Safe to cast since we know BytesLength is within the range of int.
+        memoryStream.Write(Serialization.ToBytes(this.Index));
+        memoryStream.Write(Serialization.ToBytes(this.SubIndex));
```

Comment:
I'm probably just missing it, but I can't see where these Index/SubIndex types are defined so I can't see where it is defined that this will indeed be serialized as big endian.

Reply:
The fields are defined on line 15 of this file.
These are `ulong` and uses the serialization defined in [`Helpers/Serialization.cs`](https://github.com/Concordium/concordium-net-sdk/blob/main/src/Helpers/Serialization.cs).